### ISSUE TITLE
Cut fasten-core in two modules depending on the target use

### ISSUE THREAD

**Issue by tmortagne** [fasten-project/fasten#202]
Currently, `fasten-core` include and depends on many things, many things which are useless and can create conflicts when used as a library for local analysis.

It would help a lot if all local analysis related tool (LocalMarger, DirectGrpah, etc.) were moved in a separate module on which `fasten-core` would depend. Then things like the Maven plugin and javacg-opal could depend on the new module instead of `fasten-core`.

**Comment by proksch** [comment]
I completely agree... I do not like the fact at all, that `core` contains many executables. `core` should be for data-structures and closely-related utils only, everything that gets executed should be moved to separate plugins. As of now, `core` has 167 (transitive) dependencies, having such a monster of a library makes it VERY painful to include it in other projects.
